src/server/cdn-controller.js

```diff
@@ -19,24 +22,26 @@ class CdnController {
   }
 
   registerApi(router) {
-    router.get('/api/v1/cdn/objects', [needsPermission(permissions.VIEW_FILES), jsonParser], async (req, res) => {
+    router.get('/api/v1/cdn/objects', [needsPermission(permissions.VIEW_FILES), jsonParser, validateQuery(getObjectsQuerySchema)], async (req, res) => {
       const prefix = req.query.prefix;
       const recursive = parseBool(req.query.recursive);
       const objects = await this.cdn.listObjects({ prefix, recursive });
       return res.send({ objects });
     });
 
-    router.post('/api/v1/cdn/objects', [needsPermission(permissions.CREATE_FILE), multipartParser.array('files')], async (req, res) => {
+    router.post('/api/v1/cdn/objects', [needsPermission(permissions.CREATE_FILE), multipartParser.array('files'), validateBody(postObjectsBodySchema)], async (req, res) => {
       if (req.files && req.files.length) {
-        const uploads = req.files.map(file => {
-          const fileName = urls.concatParts(req.body.prefix, file.originalname);
+        const uploads = req.files.map(async file => {
+          const fileName = req.body.prefix ? urls.concatParts(req.body.prefix, file.originalname) : file.originalname;
           const uniqueFileName = fileNameHelper.makeUnique(fileName);
-          return this.cdn.uploadObject(uniqueFileName, file.path, {});
+          await this.cdn.uploadObject(uniqueFileName, file.path, {});
```

Comment:
Why does this look to me like a breaking change? Shouldn't it return some 201 and an id to the resource?

Reply:
It only looks like a breaking change, if you look a few lines further you see that the results are discarded, that's why I changed it to make it clearer.

Technically it should return 201 and IDs (and even more technically a location header), no one will use those though, it would also be a different task IMHO.